test/parallel/test-https-close.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const common = require('../common');
+const fs = require('fs');
+const assert = require('assert');
+
+if (!common.hasCrypto) {
+  console.log('1..0 # Skipped: missing crypto');
+  return;
+}
+var https = require('https');
+
+var options = {
+  key: fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
+  cert: fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem')
+};
+
+var serverCloseEventCalled = false;
+var connections = {};
+
+var server = https.createServer(options, function(req, res) {
+  var interval = setInterval(function() {
+    res.write('data');
+  }, 5100);
+  interval.unref();
+});
+
+server.on('connection', function(connection) {
+  var key = connection.remoteAddress + ':' + connection.remotePort;
+  connection.on('close', function() {
+    delete connections[key];
+  });
+  connections[key] = connection;
+});
+
+function shutdown() {
+  server.close(function() {
+    serverCloseEventCalled = true;
```

Comment:
Nit: Instead of using the `serverCloseEventCalled` boolean, use `common.mustCall()` on the `close` callback.